src/core/controllers/stream/users/users.ts

```diff
@@ -0,0 +1,51 @@
+import * as Core from '@/core';
+import * as Config from '@/config';
+
+/**
+ * Stream User Controller
+ *
+ * Handles user stream requests from the UI layer.
+ * Coordinates between application layer and manages background data fetching.
+ * Supports followers, following, friends, and other user stream types.
+ */
+export class StreamUserController {
+  private constructor() {}
+
+  /**
+   * Get or fetch a slice of a user stream (followers, following, friends, etc.)
+   *
+   * @param streamId - User stream identifier
+   * @param user_id - ID of the user whose stream is being fetched
+   * @param skip - Number of users to skip (for pagination)
+   * @param limit - Number of users to return
+   * @returns Next page of user IDs and pagination offset
+   */
+  static async getOrFetchStreamSlice({
+    streamId,
+    user_id,
```

Comment:
Depending on whether we are on /profile (current user) dont send user_id, if we are viewing someone else's profile then do include user_id